### DISCUSSION TITLE
How to dynamically create a sub-model according to the existing value of parent model.

### DISCUSSION THREAD

**Discussion by neverrop** [pydantic/pydantic#4716] [Question]
Say I have a model A=(a=[], b=None), I want to generate A.b dynamically according to the value of A.a
such as:
```
from pydantic import BaseModel

class A(BaseModel):
    a: List[tuple] = [('obj', str), (num, int)]

    # here I want b to be dynamically generated according to A.a
    b: B = B()

# here is what A.b supposed to be according to current A.a
class B(BaseModel):
    obj: str
    num: int

```

**Comment by samuelcolvin** [comment]
If you want the **type** of `B` to be dynamic, that's not really possible and almost certainly a bad idea.

If you want the **value** of `b` to be dynamic, that should be achievable using either a property/function, or using a [validator](https://pydantic-docs.helpmanual.io/usage/validators/).
